### PR TITLE
Introduce ElasticsearchJavaBasePlugin to simplify YamlRestTestPlugin

### PR DESCRIPTION
**Pull request by breskeby**
just configuring common conventions on java based projects without
adding opinionated sourcesets. Reduces the configuration overhead for
yaml rest test only projects.

In the end we create less tasks and configure less for test only
projects.

### CODE REVIEW

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaPlugin.java [hunk 1]
```diff
@@ -52,28 +50,20 @@
...
-        // make sure the global build info plugin is applied to the root project
-        project.getRootProject().getPluginManager().apply(GlobalBuildInfoPlugin.class);
-        // common repositories setup
-        project.getPluginManager().apply(RepositoriesSetupPlugin.class);
+        project.getPluginManager().apply(ElasticsearchJavaBasePlugin.class);
         project.getPluginManager().apply(JavaLibraryPlugin.class);
-        project.getPluginManager().apply(ElasticsearchTestBasePlugin.class);
-        project.getPluginManager().apply(PrecommitTaskPlugin.class);
 
         configureConfigurations(project);
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaPlugin.java:62]: At one point we should be able to move this to the base plugin too, but turns out we have version conflicts in some projects to fix first. Fixing these was out of scope here.

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaPlugin.java [hunk 2]
```diff
@@ -111,119 +96,57 @@ public static void configureConfigurations(Project project) {
             configuration.resolutionStrategy(ResolutionStrategy::failOnVersionConflict);
         });
 
-        // force all dependencies added directly to compile/testImplementation to be non-transitive, except for ES itself
```
Comment (breskeby) [inline build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaPlugin.java:114]: that was never used

**Review by pugnascotia — APPROVED** [review]
LGTM but you'd better get a 👍 from Mark too 😁

File: build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaPlugin.java (hunk 2 above)
Comment (mark-vieira) [reply build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/ElasticsearchJavaPlugin.java:114]: I guess this is handled by the bit below?

**Review by mark-vieira — APPROVED** [review]
LGTM. :+1: for less superfluous stuff added to QA projects.

**Pull request merged**